src/base/DeltaResolver.sol

```diff
@@ -1,17 +1,33 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity ^0.8.24;
 
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
 import {Currency} from "@uniswap/v4-core/src/types/Currency.sol";
 import {ImmutableState} from "./ImmutableState.sol";
+import {TransientStateLibrary} from "@uniswap/v4-core/src/libraries/TransientStateLibrary.sol";
 
 /// @notice Abstract contract used to sync, send, and settle funds to the pool manager
 /// @dev Note that sync() is called before any erc-20 transfer in `settle`.
 abstract contract DeltaResolver is ImmutableState {
+    using TransientStateLibrary for IPoolManager;
+
+    /// @notice Value used to signal that an entire open delta should be taken/settled
+    uint256 public constant ENTIRE_OPEN_DELTA = 0;
+
+    /// @notice Emitted trying to settle a positive delta, or take a negative delta
+    error InvalidDeltaForAction();
+
     /// @notice Take an amount of currency out of the PoolManager
     /// @param currency Currency to take
     /// @param recipient Address to receive the currency
     /// @param amount Amount to take
     function _take(Currency currency, address recipient, uint256 amount) internal {
+        if (amount == ENTIRE_OPEN_DELTA) {
```

Comment:
I do think in general I'd prefer this logic to be handled outside of this library. I view DeltaResolver as a pretty agnostic library that is really just used to share code for the direct poolManager calls to `take`, `settle`, `mint`, and `burn`.

Reply:
Also I think you and I have been in general MAXing not having to use booleans to do certain behavior which is why I like "close" it just knows you want to settle the whole delta. If we move to this we're checking booleans/parameters in commands. id just rather have more commands that do specific behaviors

Reply:
yeah thats fair, i've added a new command. I think routing will want generic TAKE and also TAKE_PORTION. But for now implemented TAKE_ALL for the generic swap case